superset/assets/javascripts/SqlLab/components/TabbedSqlEditors.jsx

```diff
@@ -28,34 +29,54 @@ let queryCount = 1;
 class TabbedSqlEditors extends React.PureComponent {
   constructor(props) {
     super(props);
-    const uri = window.location.toString();
-    const search = window.location.search;
-    const cleanUri = search ? uri.substring(0, uri.indexOf('?')) : uri;
-    const query = search.substring(1);
+    const sqlLabUrl = '/superset/sqllab';
     this.state = {
-      uri,
-      cleanUri,
-      query,
+      sqlLabUrl,
+      query: null,
       queriesArray: [],
       dataPreviewQueries: [],
       hideLeftBar: false,
     };
   }
-  componentWillMount() {
-    if (this.state.query) {
-      queryCount++;
-      const queryEditorProps = {
-        title: getParamFromQuery(this.state.query, 'title'),
-        dbId: parseInt(getParamFromQuery(this.state.query, 'dbid'), 10),
-        schema: getParamFromQuery(this.state.query, 'schema'),
-        autorun: getParamFromQuery(this.state.query, 'autorun'),
-        sql: getParamFromQuery(this.state.query, 'sql'),
-      };
-      this.props.actions.addQueryEditor(queryEditorProps);
-      // Clean the url in browser history
-      window.history.replaceState({}, document.title, this.state.cleanUri);
+  componentDidMount() {
+    const search = window.location.search;
+    if (search) {
+      const queryString = search.substring(1);
+      const urlId = getParamFromQuery(queryString, 'id');
+      const self = this;
+      if (urlId) {
+        $.ajax({
+          type: 'GET',
+          url: `/kv/${urlId}`,
+          success(data) {
```

Comment:
you could get rid of 
`const self = this;`

if you use a fat arrow method for `success`

```js
if (urlId) {
  $.ajax({
    type: 'GET',
    url: `/kv/${urlId}`,
    success: (data) => {
      this.popNewTab(JSON.parse(data));
    },
  });
}
```